auctions/dutch-auction/src/abi.sw

```diff
@@ -0,0 +1,11 @@
+library abi;
+
+use std::{contract_id::ContractId, identity::Identity};
+
+abi DutchAuction {
```

Comment:
I think this needs another function. I'm unsure. How will the frontend know which auction to show to users? Iterating over an infinite number of auctions in the mapping and checking that they have not ended (and currently exist) is inefficient and even then there's nothing to indicate which auction should be shown and to who.

Reply:
After vectors are added we can have an "active auctions" vector in storage perhaps

Reply:
when vectors get added we can just have a vector of currently running auctions

Reply:
I'm doing something similar in the Escrow (using an array for now but planning on using vec) however I dislike it because the amount of juggling to get data in / out might be a pain. I'd prefer to use maps in some intelligent way to have instant (O(1)) access. We'll see what happens when vec is out

Reply:
You mean like move the ended auctions into a seperate storagemap maybe? I dont think anything like thats possible right now though.

Added the auction() function btw

Reply:
My understanding is that you want to store the currently active auctions in a vector. This vector will increase in size as more auctions are added. You need to somehow know how to index into the vector to get the auction that you want because iterating over the vector is unacceptable. It will eventually get too large to iterate and you'll deadlock the contract. This means we need to know how to index into the vector via a mechanism similar to the storage map. Another problem is then removing the active auctions because that will reallocate a lot of memory which is something that we do not want to do therefore using a vector at all is a bad idea because even if we never remove from the vector eventually adding to it will make it reallocate so much memory that it will deadlock the contract again.
This leaves storage mappings as the only option but now you need to figure out how that can be used effectively. I have not thought that through yet...

Reply:
You could store a counter for active auctions and then just have a 0..n mapping of active auctions, and moving all the auctions down by one whenever one ends, but thats just an emulation of a vector, so sounds like a pointless endeavor. Other than that I have no idea what you could do with storage maps here that would allow you to track only the active auctions and remove the ones that end